### PR TITLE
Added note about cve-2024-22030 to security faq

### LINKED ISSUE #1175: Update documentation around CVE-2024-22030
## Related Issues
[SURE-7667]

## Summary

Documentation includes two phases: 

- Update exist docs around our concerns, for example: We don't properly document our assumptions around CA validations and the risks associated with it (Rancher's threat model for at least certs/CA/TLS/SSL) in our docs.
- Update docs around any changes in behavior, or functionality. Mainly with regard to cert rotation, and changes with agent or flags. 

Resources: 
- https://ranchermanager.docs.rancher.com/reference-guides/rancher-security/security-advisories-and-cves
- https://www.suse.com/c/rancher-security-update/?_gl=1*5lvo5u*_ga*NzM4OTQzOTk4LjE3MDk1Njg3NjU.*_ga_Y7SFXF9L00*MTcxMDQ0Nzk4Ny4xNC4wLjE3MTA0NDc5ODguNTkuMC4w

### PR DESCRIPTION
**Pull request by martyav**
Addresses part 1 of #1175 

## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description

There was a request to update the docs in respect to a new CVE. 

> We don't properly document our assumptions around CA validations and the risks associated with it (Rancher's threat model for at least certs/CA/TLS/SSL) in our docs.

This PR attempts to address that part of the request. I had some trouble figuring out where to place the content, but the Security FAQ seems as good as any. I decided not to include the advice from the linked KB article, but I'm not opposed to including it under a best practices or security section. 

A later step will be to update the docs if any changes occur to defend against the exploit. That (any new defenses added to Rancher) is still being worked on.

I also updated the headings so that the page will now have a right sidebar.

### CODE REVIEW

File: docs/faq/security.md [hunk 1]
```diff
@@ -7,12 +7,16 @@ title: Security FAQ
...
 
-**Is there a Hardening Guide?**
+### Is there a Hardening Guide?
 
 The Hardening Guide is now located in the main [Security](../reference-guides/rancher-security/rancher-security.md) section.
 
-<br/>
-
-**What are the results of Rancher's Kubernetes cluster when it is CIS benchmarked?**
+### What are the results of Rancher's Kubernetes cluster when it is CIS benchmarked?
```
Comment (martyav) [inline docs/faq/security.md:14]: This heading is written kind of awkwardly but maybe we can look into fixing that in another PR. I did notice it but couldn't think of a decent way to rephrase

Comment (sunilarjun) [inline docs/faq/security.md:14]:
```suggestion
### Where can I access results after running a CIS benchmark on a hardened Rancher Kubernetes cluster?
```

Comment (sunilarjun) [reply docs/faq/security.md:14]: Just a suggestion but perhaps something like above to rephrase the header?

File: docs/faq/security.md [hunk 2]
```diff
@@ -7,12 +7,16 @@ title: Security FAQ
...
+
+As described in [Rancher Security Update CVE-2024-22030](https://www.suse.com/c/rancher-security-update/), under a narrow set of circumstances, malicious actors can take over Rancher nodes, by exploiting the behavior of Rancher CAs. For the attack to succeed, the malicious actor must generate a valid certificate from either a valid CA in the targeted Rancher server, or from a valid registered CA. The attacker also needs to either hijack or spoof the Rancher server-url as a preliminary step. Rancher is currently evaluating Rancher CA behavior to mitigate against this and any similar avenues of attack.
```
Comment (sunilarjun) [inline docs/faq/security.md:22]:
```suggestion
As described in [Rancher Security Update CVE-2024-22030](https://www.suse.com/c/rancher-security-update/), under a narrow set of circumstances, malicious actors can take over Rancher nodes by exploiting the behavior of Rancher CAs. For the attack to succeed, the malicious actor must generate a valid certificate from either a valid CA in the targeted Rancher server, or from a valid registered CA. The attacker also needs to either hijack or spoof the Rancher server-url as a preliminary step. Rancher is currently evaluating Rancher CA behavior to mitigate against this and any similar avenues of attack. 
```

Comment (sunilarjun) [reply docs/faq/security.md:22]: Just a nitpick but seemed like an extra comma here, otherwise looks good.

**Review by sunilarjun — COMMENTED** [review]
Left a rephrase suggestion and a small grammar adjustment, otherwise looks good. Thanks Marty!

File: docs/faq/security.md (hunk 1 above)
Comment (martyav) [reply docs/faq/security.md:14]: Thanks for giving it a try. I am having a hard time with this heading title and think I'll shelve it for a later PR. Maybe the point it's trying to get across is that we have self-assessment guides in the Security section of the docs? @btat could you weigh in?

**Review by sunilarjun — APPROVED** [review]

File: docs/faq/security.md (hunk 1 above)
Comment (martyav) [inline docs/faq/security.md:14]:
```suggestion
### Have hardened Rancher Kubernetes clusters been evaluated by the CIS Kubernetes Benchmark? Where can I find the results?
```

Comment (martyav) [reply docs/faq/security.md]: I committed @btat's suggestion to the branch

**Review by pjbgf — APPROVED** [review]

**Review by macedogm — APPROVED** [review]

**Review by LucasSaintarbor — APPROVED** [review]

**Review by btat — APPROVED** [review]

**Pull request merged**